events/views.py

```diff
@@ -30,32 +31,41 @@ def get(self,request,year):
         data = serializer.data
         return Response({"message":"Events Fetched successfully","data":data},status.HTTP_200_OK)
 
-# TODO: simplify with drf
-@api_view(['POST', ])
-@ecell_user
-def event_register(request, id):
-    eventregister = EventRegister()
-    user = request.ecelluser
-    res_status = status.HTTP_401_UNAUTHORIZED
-    if user.verified:
-        eventregister.user = user
+
+class EventRegisterView(APIView):
+    permission_classes = [IsAuthenticated]
+    @swagger_auto_schema(
+        operation_id="event_register",
+        responses = {
+            '201': set_example(responses.event_registration_201),
+            '200':set_example(responses.user_already_registered_event_200),
+            '404':set_example(responses.event_does_not_exist_404),
+            '401':set_example(responses.user_unauthorized_401)
+        }
+    )
+    def post(self,request,id):
+        user = request.user
+        eventregister = EventRegister()
+        event = Event.objects.all()
```

Comment:
I did not understand why this is here. Is there a necessity?
```suggestion
```

Reply:
done sir